app/jobs/scheduled/delete_old_chat_messages.rb

```diff
@@ -12,23 +12,35 @@ def execute(args = {})
     def delete_public_channel_messages
       return unless valid_day_value?(:chat_channel_retention_days)
 
-      ChatMessage
+      destroyed_ids = ChatMessage
```

Comment:
This should probably be done in batches... otherwise we could end up trying to load 10000s of messages into memory in one go. 

(but I realise that problem was not introduced by this PR, so feel free to ignore for now)

Reply:
probably a good time to fix it too though... Given it's ran every day, maybe we limit to 10000 and do a find_each ?

Reply:
Sounds good to me :+1: Even with find_each, make sure it's not run inside a transaction because that'll cause ActiveRecord to keep all the modified models in memory. (I had that fun with https://github.com/discourse/discourse/commit/67b23c0e22ed851ec7924709b53e416356980682)